### DISCUSSION TITLE
Tracking crossing zone with stationary objects detection

### DISCUSSION THREAD

**Discussion by brainstormi** [blakeblackshear/frigate#15303] [Ask A Question]
I'm trying to create some automations when my car enters and leaves the garage. This parking place is like a corridor, where 2 cars can park one behind the other. For that I created three zones:

1. car in ramp (just before the door).
2. car entered in (zone where it would be safe already to close the door, it would be a subset of the next zone)
3. car parking (all the parking place where the 2 cars stayed parked)

The issue is trying to track when the cars get out of the parking with stationary objects detection. When I track the mqtt frigate events I see that that 'entered_zones' field doesn't change when the car start moving to get out. It always reflect the original zones order when the car was detect when entering the garage. Only 'current_zone' changes dinamically. Do you have any idea about how to define such automation effectively?

**Comment by NickM-27** [comment]
what exactly are you trying to accomplish? can you not look at current zone to know when the object is no longer in the zone?

**Reply by brainstormi** [reply]
Basically trying to get the direction of the car, if it's going out or in to open the garage door or close it following specific rules depending on the case. 
When the car is entering the garage it's easier because the object it's tracked for first time and I can play with 'entered_zones'. For car going out not, because tracking stationary objects the 'entered_zones' field doesn't work anymore for this purpose, it contains all the zones and in the same order when the car entered the garage before. So I was looking if there was any more 'elegant' way for tracking it when moving again to detect movement direction across the zones or playing with 'current_zone' as you suggested is the only way to go.

**Reply by NickM-27** [reply]
It would be a lot easier to suggestion a solution if you can describe the zones and the usual path, but I don't see any reason this can't work. 

You could just check `before` for entered_zones having the inner zone and `after.current_zones` including the outer zone, that way you know it is currently leaving

**Reply by brainstormi** [reply]
Playing a bit with your approach suggested, the issue now is detecting multiple times the same event meanwhile the car is detected in the "current_zones"

**Reply by NickM-27** [reply]
what is your current logic? sounds like you might not be checking before to make sure this is the first time

**Reply by brainstormi** [reply]
![image](https://github.com/user-attachments/assets/9c1ef298-9d70-4dd1-8bc1-c2e1b71c162f)

If I use a condition 'before' for coche_parking zone in current zones, doesn't work, because when detected first in 'coche_rampa', before appears empty. These 2 zones are consecutive, maybe I would need to overlap them a bit to avoid this gap in the detection.

**Reply by NickM-27** [reply]
it seems like these are reversed, you should apply the `not in parking` to the before entered zones
